File: framework/languages/dotnet/samples/TicTacToe/Server/Play/Actors/PlayActorFactory.cs
namespace TicTacToe.Server.Play.Actors;

internal sealed class PlayActorFactory(ILogger<PlayActor> logger) : IZLinkActorFactory
{
    public ValueTask<IZLinkActor> CreateAsync(
        string actorId,
        IZLinkActorContext context,
        CancellationToken cancellationToken = default)
    {
        cancellationToken.ThrowIfCancellationRequested();
        return ValueTask.FromResult<IZLinkActor>(new PlayActor(actorId, context, logger));
    }
}

File: framework/languages/node/samples/TicTacToe/Server/Play/Actors/play-actor-factory.js
const { PlayActor } = require('./play-actor');
const { actorDisplayName } = require('../../../Shared/Contracts/messages');

class PlayActorFactory {
  constructor() {
    this.actors = new Map();
  }

  ensure(actorId) {
    if (!this.actors.has(actorId)) {
      this.actors.set(actorId, new PlayActor(actorId, actorDisplayName(actorId)));
    }
    return this.actors.get(actorId);
  }
}

module.exports = { PlayActorFactory };
